packages/playwright-core/src/remote/playwrightServer.ts

```diff
@@ -96,6 +98,23 @@ export class PlaywrightServer {
         headers.push(process.env.PWTEST_SERVER_WS_HEADERS!);
       });
     }
+    server.on('upgrade', (request, socket, head) => {
+      const pathname = url.parse(request.url!).pathname;
```

Comment:
```suggestion
      const pathname = new URL(request.url!).pathname;
```

Let's try to avoid url module?